test/notebook-tests.py

```diff
@@ -63,6 +46,8 @@
 
 
 def test_t1():
+    import kso_utils.widgets as kso_widgets
```

Comment:
Why did you move all the imports to inside the test functions and not just at the top of the document?

Reply:
I find it useful to have the imports listed that are in used in each test. That way we can easily see what is needed. If there are functions which are common to all tests, we could move them out.